From 707857ae9fef23a4f292dfcb60720a5496d6939a Mon Sep 17 00:00:00 2001
Message-Id: <707857ae9fef23a4f292dfcb60720a5496d6939a.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Mon, 2 Feb 2015 14:53:38 +0530
Subject: [PATCH 10/61] MUST_REBASE [VPG]: drivers/video/adf: Fix sporadic
 screen shift on DSI

This patches fixes the below issues:
     1. display off sequence needs to be send after shutdown packet
	otherwise sporadically split screen issue may occur
     2. Add wait for dsi fifo to be empty after sending the display off
	sequence for DSI

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5611
Change-Id: I326890542c453d01c5ef0a1781407dfe6352b3b3
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    7 -------
 drivers/video/adf/intel/core/vlv/vlv_pipeline.c    |    9 +++++++++
 2 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index 78075e43..158892f 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -285,13 +285,6 @@ static int dsi_display_off(struct intel_pipe *pipe)
 		goto out;
 	}
 
-	/*
-	 * if disable packets are sent before sending shutdown packet then in
-	 * some next enable sequence send turn on packet error is observed
-	 */
-	if (panel->ops->power_off)
-		panel->ops->power_off(dsi_pipe);
-
 	vlv_post_pipeline_off(pipeline);
 
 	if (panel->ops->disable_panel_power)
diff --git a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
index 6ee50cb..8facbd4 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
@@ -453,11 +453,13 @@ int vlv_dsi_post_pipeline_off(struct intel_pipeline *pipeline)
 	struct vlv_dsi_port *dsi_port = NULL;
 	struct dsi_pipe *dsi_pipe = NULL;
 	struct dsi_context *dsi_ctx = NULL;
+	struct dsi_panel *panel = NULL;
 	enum port port;
 	u32 err = 0;
 
 	dsi_pipe = &disp->gen.dsi;
 	dsi_ctx = &dsi_pipe->config.ctx;
+	panel = dsi_pipe->panel;
 
 	if (vlv_is_vid_mode(pipeline)) {
 		/* Send Shutdown command to the panel in LP mode */
@@ -472,6 +474,13 @@ int vlv_dsi_post_pipeline_off(struct intel_pipeline *pipeline)
 		goto out;
 	}
 
+	/*
+	 * if disable packets are sent before sending shutdown packet then in
+	 * some next enable sequence send turn on packet error is observed
+	 */
+	if (panel->ops->power_off)
+		panel->ops->power_off(dsi_pipe);
+
 	for_each_dsi_port(port, dsi_ctx->ports) {
 		dsi_port = &disp->port.dsi_port[port];
 		vlv_dsi_port_wait_for_fifo_empty(dsi_port);
-- 
1.7.9.5

